basic/atm/atm.py

```diff
@@ -0,0 +1,111 @@
+import sys
+
+CUSTOMERS_FILE_LOCATION = 1
+NAME_INDEX = 0
+PASSWORD_INDEX = 1
+BALANCE_INDEX = 2
+ID_INDEX = 3
+CUSTOMER_LINE_FORMAT = '{};{};{};{}\n'
+WELCOME_MESSAGE = 'Welcome to the ATM! \n ' \
+                  'To exit enter -1, ' \
+                  'otherwise please enter your customer ID:'
+MENU_MESSAGE = 'To check your balance, press 1 \n' \
+               'To make a deposit, press 2 \n' \
+               'To make a withdrawal, press 3 \n' \
+               'To change your password, press 4 \n '
+
+
+def get_customers(file):
+    customers = {}
+    for line in file.readlines():
+        data = line.strip().split(';')
+        customers[data[ID_INDEX]] = [data[NAME_INDEX],
+                                     data[PASSWORD_INDEX],
+                                     int(data[BALANCE_INDEX])]
+    return customers
+
+
+def check_balance(customers, ID):
+    print('Your balance is: {}'.format(customers[ID][BALANCE_INDEX]))
+
+
+def deposit(customers, ID):
+    try:
+        amount = int(input('Enter amount to deposit: '))
+        if amount > 0:
+            customers[ID][BALANCE_INDEX] += amount
+        else:
+            print('Please enter a positive number')
+    except ValueError:
+        print('Please enter an integer')
+
+
+def withdraw(customers, ID):
+    try:
+        amount = int(input('Enter amount to withdraw: '))
+        if amount > 0:
+            customers[ID][BALANCE_INDEX] -= amount
+        else:
+            print('Please enter a positive number')
+    except ValueError:
+        print('Please enter an integer')
+
+
+def change_password(customers, ID):
+    tries = 0
+    while tries < 3:
+        password = input('Enter old password: ')
+        if password == customers[ID][PASSWORD_INDEX]:
+            customers[ID][PASSWORD_INDEX] = input('Enter new password: ')
+            break
+        else:
+            print('Incorrect password, try again')
+            tries += 1
+    if tries == 3:
+        print('Exceeded number of tries allowed')
+
+
+def update_file(file, customers):
+    file.seek(0)
+    file.truncate()
+    for key in customers:
+        file.writelines(
+            CUSTOMER_LINE_FORMAT.format(customers[key][NAME_INDEX],
+                                        customers[key][PASSWORD_INDEX],
+                                        customers[key][BALANCE_INDEX],
+                                        key))
+
+
+def main():
+    with open(customers_file_path, 'r+') as customers_file:
```

Comment:
the file should be opened only when you need to read/write to it and not to the whole time